### PR TITLE
Fix integration test suite

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Fixes the integration test suite

## 🔗 Related links

- [Asana Task](https://app.asana.com/0/1202805690238892/1203288271928429/f) _(internal)_

## 🔍 What does this change?

- Adjust the integration test suite to pass
- Slightly changes the deduplication logic on joining tables by setting required tables as compiler artifact. Before, it was possible that in some edge cases, a table, which was selected from, were deduplicated.

## 🛡 What tests cover this?

With this PR, all Rust tests are passing. This includes the unit tests, integration tests, *and* REST tests.

### CODE REVIEW

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/entity_type.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ impl Path for EntityTypeQueryPath {
                         column: "target_property_type_version_id",
                     },
                     join_table_name: TableName::PropertyTypes,
-                    join_column_access: DataTypeQueryPath::VersionId.column_access(),
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/store/postgres/query/entity_type.rs:51]: Was this just a typo/mistake before? :)

File: packages/graph/hash_graph/tests/integration/postgres/links.rs [hunk 2]
```diff
@@ -234,44 +234,32 @@ async fn ordered_links() {
...
-            person_c_metadata.identifier().entity_id(),
-            friend_link_type_id.clone(),
-            1,
-        )
-        .await
-        .expect("could not create link");
+    assert!(
+        !api.get_entity_links(person_a_metadata.identifier().entity_id())
+            .await
+            .expect("could not fetch link")
```
Comment (thehabbos007) [inline packages/graph/hash_graph/tests/integration/postgres/links.rs:251]: Should this be the following as we're returning multiple?
```suggestion
            .expect("could not fetch links")
```

File: packages/graph/hash_graph/tests/integration/postgres/links.rs [hunk 3]
```diff
@@ -234,44 +234,32 @@ async fn ordered_links() {
...
-        ][..],
-        links_from_source
-            .iter()
-            .map(|link| link.inner().target_entity())
-            .collect::<Vec<_>>()[..]
-    )
+    assert!(
+        api.get_entity_links(person_a_metadata.identifier().entity_id())
+            .await
+            .expect("could not fetch link")
```
Comment (thehabbos007) [inline packages/graph/hash_graph/tests/integration/postgres/links.rs:262]: Like above, should this be the following as we're returning multiple?
```suggestion
            .expect("could not fetch links")
```

**Review by thehabbos007 — COMMENTED** [review]
Thank you Tim! I have some comments :)

File: packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/mod.rs [hunk 4]
```diff
@@ -231,7 +231,7 @@ impl<C: AsClient> EntityStore for PostgresStore<C> {
 
     async fn get_entity<'f: 'q, 'q>(
         &self,
-        query: &'q StructuralQuery<'q, Entity>,
+        query: &'f StructuralQuery<'q, Entity>,
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/mod.rs:234]: `'f` may outlive `'q` here as the reference to the query does not need to live as long as the filter itself. Last time I tried removing one of the lifetimes it failed.

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs [hunk 5]
```diff
@@ -351,29 +354,33 @@ impl<'f: 'q, 'q, T: PostgresQueryRecord<'q>> SelectCompiler<'f, 'q, T> {
...
-                if let Some(last_join) = self.statement.joins.last_mut() {
-                    if last_join.join == current_column {
-                        last_join.join.table.name = join_column.table.name;
-                        last_join.join.access = join_column.access;
-                        current_table = last_join.join.table;
+                if !self
+                    .artifacts
+                    .required_tables
+                    .contains(&last_join.join.table)
+                    && last_join.join == current_column
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs]: Good point: befbabe772846576e0d289f6ea1e6a402a140fb0

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/entity_type.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/entity_type.rs:51]: Yes. The result is the same, but using `PropertyTypeQueryPath` is clearer as the `Table` also is `PropertyTypes`. In the long run we should probably also have enumerations for columns instead, so we can use something like `PropertyTypesColumn::VersionId`.

**Comment by TimDiekmann** [comment]
Thanks @thehabbos007, I applied your suggestions 🙂

**Review by thehabbos007 — APPROVED** [review]
Thank you :sparkles:
